lib/Site.js

```diff
@@ -125,7 +125,7 @@ function getGlyphiconsMap() {
   const glyphiconsClasses = fs.readFileSync(glyphiconsPath, 'utf8').trimRight().split(/\r?\n/);
   return glyphiconsClasses.reduce((glyphiconsMap, glyphiconClass) => {
     const name = glyphiconClass.replace(/-/g, '_');
-    const html = `<span class='glyphicon ${glyphiconClass}' aria-hidden='true'></span>`;
+    const html = `<span class='glyphicon ${glyphiconClass}' aria-hidden='true'>&#8203</span>`;
```

Comment:
I think you might want to create a new issue for this in order to explain why this change is necessary.

Reply:
This "bug" appeared when my implementation used `Cheerio` after `markbinder.renderFile`. It wouldn't have appeared otherwise.

Will clarify further in commit message.

Reply:
Must it be done after `markbinder.renderFile`?

Reply:
As not all the html elements / tags has been added to the pageData, I did not attempt to do it before .renderFile.  

Will experiment tomorrow.

Reply:
- Why does it need all the html elements / tags to have been added?
- Can we assume/limit that footer can only be in addressable pages?
- Can I clarify that footer must be explicitly included?

Reply:
- Shall we specify `footer` in front matter similar to what [Jekyll](https://jekyllrb.com/docs/frontmatter/) does with `layout`?
  - This allows separation from the content.
    ```html
    <frontmatter>
    title: MarkBind
    keywords: mark, markdown, markbind, fragment, reusable, CLI
    footer: footer.md
    </frontmatter>
    ```
  - Let's require file extension ([Assemble](http://assemble.io/docs/YAML-front-matter.html) does too).
  - We can assume it's relative to `_markbind/layouts`.

Reply:
Moved to https://github.com/MarkBind/markbind/pull/238#issuecomment-391683933 for extended discussion.